src/libmongoc/tests/unified/entity-map.c

```diff
@@ -771,6 +660,7 @@ entity_client_new (entity_map_t *em, bson_t *bson, bson_error_t *error)
    }
 
    if (can_reduce_heartbeat && em->reduced_heartbeat) {
+      // @todo This option is needed for both single-threaded and pooled clients, but only works on pooled
```

Comment:
`MONGOC_URI_HEARTBEATFREQUENCYMS` applies to single-threaded too. Possibly related: single-threaded has a non-configurable [5 second cooldown](https://github.com/mongodb/specifications/blob/787bbe67bdffb614270654372112e93fc2cd3eb3/source/server-discovery-and-monitoring/server-monitoring.md#cooldownms) blocks prevent repeated connection attempts after failure to the same server.

This can be bypassed in testing with the private `_mongoc_topology_bypass_cooldown`. For example, this test prints heartbeat attempts every 500ms:

```c
static void
print_heartbeat_started (const mongoc_apm_server_heartbeat_started_t *event)
{
   const mongoc_host_list_t *host = mongoc_apm_server_heartbeat_started_get_host (event);
   MONGOC_DEBUG ("heartbeat started to: %s\n", host->host_and_port);
}
static void
test_max_heartbeat_frequency_ms (void)
{
   // Specify a non-listening port:
   mongoc_uri_t *uri = mongoc_uri_new ("mongodb://localhost:12345");
   // Disable default single-threaded try-once so scanning repeats until timeout:
   mongoc_uri_set_option_as_bool (uri, MONGOC_URI_SERVERSELECTIONTRYONCE, false);
   // Set heartbeat frequency to the minimum possible:
   mongoc_uri_set_option_as_int32 (uri, MONGOC_URI_HEARTBEATFREQUENCYMS, 500);
   mongoc_client_t *client = mongoc_client_new_from_uri (uri);
   // Bypass single-threaded cooldown to speed up retry attempts:
   _mongoc_topology_bypass_cooldown (client->topology);
   // Add a callback to print every time a heartbeat is attempted:
   {
      mongoc_apm_callbacks_t *cbs = mongoc_apm_callbacks_new ();
      mongoc_apm_set_server_heartbeat_started_cb (cbs, print_heartbeat_started);
      mongoc_client_set_apm_callbacks (client, cbs, NULL);
      mongoc_apm_callbacks_destroy (cbs);
   }
   bson_error_t error;
   bool ok = mongoc_client_command_simple (client, "admin", tmp_bson ("{'ping': 1}"), NULL, NULL, &error);
   ASSERT_WITH_MSG (!ok, "expected command to fail, got success");
   ASSERT_ERROR_CONTAINS (
      error, MONGOC_ERROR_SERVER_SELECTION, MONGOC_ERROR_SERVER_SELECTION_FAILURE, "No suitable servers found");
   mongoc_client_destroy (client);
   mongoc_uri_destroy (uri);
}
```

Reply:
Ok thanks, I'll remove the comment.